framework/src/plugins/base_plugin.ts

```diff
@@ -295,64 +284,58 @@ export abstract class BasePlugin<
 // TODO: Once the issue fixed we can use require.resolve to rewrite the logic
 //  https://github.com/facebook/jest/issues/9543
 export const getPluginExportPath = (
-	pluginKlass: InstantiablePlugin,
+	PluginKlass: InstantiablePlugin,
 	strict = true,
 ): string | undefined => {
 	let nodeModule: Record<string, unknown> | undefined;
 	let nodeModulePath: string | undefined;
 
+	const plugin = new PluginKlass();
+
 	try {
 		// Check if plugin name is an npm package
 		// eslint-disable-next-line global-require, import/no-dynamic-require, @typescript-eslint/no-var-requires, @typescript-eslint/no-unsafe-assignment
-		nodeModule = require(pluginKlass.info.name);
-		nodeModulePath = pluginKlass.info.name;
+		nodeModule = require(plugin.name);
+		nodeModulePath = plugin.name;
 	} catch (error) {
-		/* Plugin info.name is not an npm package */
+		/* Plugin pluginKlass.name is not an npm package */
 	}
 
-	if (!nodeModule && pluginKlass.info.exportPath) {
+	if (!nodeModule && plugin.nodeModulePath) {
 		try {
-			// Check if plugin name is an npm package
+			// Check if plugin nodeModulePath is an npm package
 			// eslint-disable-next-line global-require, import/no-dynamic-require, @typescript-eslint/no-var-requires, @typescript-eslint/no-unsafe-assignment
-			nodeModule = require(pluginKlass.info.exportPath);
-			nodeModulePath = pluginKlass.info.exportPath;
+			nodeModule = require(plugin.nodeModulePath);
+			nodeModulePath = plugin.nodeModulePath;
 		} catch (error) {
-			/* Plugin info.exportPath is not an npm package */
+			/* Plugin nodeModulePath is not an npm package */
 		}
 	}
 
-	if (!nodeModule || !nodeModule[pluginKlass.name]) {
+	if (!nodeModule || !nodeModule[plugin.name]) {
 		return;
 	}
 
 	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-	if (strict && nodeModule[pluginKlass.name] !== pluginKlass) {
+	if (strict && nodeModule[PluginKlass.name] !== PluginKlass) {
 		return;
 	}
 
 	// eslint-disable-next-line consistent-return
 	return nodeModulePath;
 };
 
-export const validatePluginSpec = (
-	PluginKlass: InstantiablePlugin,
-	options: Record<string, unknown> = {},
-): void => {
-	const pluginObject = new PluginKlass(options as PluginOptionsWithAppConfig);
-
-	assert(PluginKlass.alias, 'Plugin alias is required.');
-	assert(PluginKlass.info.name, 'Plugin name is required.');
-	assert(PluginKlass.info.author, 'Plugin author is required.');
-	assert(PluginKlass.info.version, 'Plugin version is required.');
-	assert(pluginObject.events, 'Plugin events are required.');
-	assert(pluginObject.actions, 'Plugin actions are required.');
+export const validatePluginSpec = (plugin: BasePlugin): void => {
+	assert(plugin.name, 'Plugin name is required.');
+	assert(plugin.events, 'Plugin events are required.');
+	assert(plugin.actions, 'Plugin actions are required.');
```

Comment:
In past we discussed to not use `assert` package any more. If that's the case we should refactor it now as we are already changing this file a lot. @shuse2

Reply:
we don't use assert anymore, but since this PR is blocking other PR and it's already pretty big, refactoring later is also fine. up to @Incede  =)